assets/js/modules/analytics/datastore/profiles.js

```diff
@@ -0,0 +1,342 @@
+					[ `${ accountId }::${ propertyId }` ]: true,
+				},
+			};
+		}
+
+		case FETCH_PROFILES: {
+			const { accountId, propertyId } = action.payload;
+
+			return {
+				...state,
+				isFetchingProfiles: {
+					...state.isFetchingProfiles,
+					[ `${ accountId }::${ propertyId }` ]: true,
+				},
+			};
+		}
+
+		case RECEIVE_CREATE_PROFILE: {
+			const { accountId, propertyId, profile } = action.payload;
+
+			return {
+				...state,
+				isDoingCreateProfile: {
+					...state.isDoingCreateProfile,
+					[ `${ accountId }::${ propertyId }` ]: false,
+				},
+				profiles: [
+					...state.profiles || [],
+					profile,
+				],
+			};
+		}
+
+		case RECEIVE_CREATE_PROFILE_FAILED: {
+			const { accountId, error, propertyId } = action.payload;
+
+			return {
+				...state,
+				error,
+				isDoingCreateProfile: {
+					...state.isDoingCreateProfile,
+					[ `${ accountId }::${ propertyId }` ]: false,
+				},
+			};
+		}
+
+		case RECEIVE_PROFILES: {
+			const { accountId, propertyId, profiles } = action.payload;
+
+			return {
+				...state,
+				isFetchingProfiles: {
+					...state.isFetchingProfiles,
+					[ `${ accountId }::${ propertyId }` ]: false,
+				},
+				profiles,
+			};
+		}
+
+		case RECEIVE_PROFILES_FAILED: {
+			const { accountId, error, propertyId } = action.payload;
+
+			return {
+				...state,
+				error,
+				isFetchingProfiles: {
+					...state.isFetchingProfiles,
+					[ `${ accountId }::${ propertyId }` ]: false,
+				},
+			};
+		}
+
+		default: {
+			return { ...state };
+		}
+	}
+};
+
+export const resolvers = {
+	*getProfiles( accountId, propertyId ) {
+		try {
+			const response = yield actions.fetchProfiles( accountId, propertyId );
+			const { profiles } = response;
+
+			yield actions.receiveProfiles( { accountId, propertyId, profiles } );
+
+			return;
+		} catch ( error ) {
+			// TODO: Implement an error handler store or some kind of centralized
+			// place for error dispatch...
+			return actions.receiveProfilesFailed( { accountId, propertyId, error } );
+		}
+	},
+};
+
+export const selectors = {
+	/**
+	 * Get all Google Analytics profiles this user account+property has available.
+	 *
+	 * Returns an array of all profiles.
+	 *
+	 * Returns `undefined` if accounts have not yet loaded.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @param {string} accountId The Analytics Account ID to fetch profiles for.
+	 * @param {string} propertyId The Analytics Property ID to fetch profiles for.
+	 * @return {Array|undefined} An array of Analytics profiles; `undefined` if not loaded.
+	 */
+	getProfiles( state, accountId, propertyId ) {
+		invariant( accountId, 'accountId is required' );
+		invariant( propertyId, 'propertyId is required' );
+
+		const { profiles } = state;
+
+		if ( profiles && profiles.length ) {
+			return profiles.filter( ( profile ) => {
+				return (
+					profile.accountId === accountId &&
+					profile.webPropertyId === propertyId
+				);
+			} );
```

Comment:
This logic shouldn't be used, profiles should be globally keyed by account ID and property ID.